ui/lib/utils.ts

```diff
@@ -161,3 +169,122 @@ export const convertImage = (image: string) => {
   //one slash docker images w/o docker.io
   return `https://hub.docker.com/r/${prefix}/${noTag}`;
 };
+
+// getSourceRefForAutomation returns the automation's sourceRef
+// depending on whether the automation is a Kustomization or a HelmRelease.
+export function getSourceRefForAutomation(
+  automation?: Automation
+): FluxObjectRef | undefined {
+  return automation?.kind === FluxObjectKind.KindKustomization
+    ? (automation as Kustomization)?.sourceRef
+    : (automation as HelmRelease)?.helmChart?.sourceRef;
+}
+
+// findNode searches for a node by the node's name and namespace.
+export function findNode(
+  nodes: FluxObjectNode[],
+  name: string,
+  namespace: string
+): FluxObjectNode | null {
+  const matches = nodes.filter(
+    (node) => node.name === name && node.namespace === namespace
+  );
+
+  if (matches.length > 0) {
+    return matches[0];
+  } else {
+    return null;
+  }
+}
+
+// getNeighborNodes returns nodes which depend on the current node
+// or are dependencies of the current node.
+export function getNeighborNodes(
+  nodes: FluxObjectNode[],
+  currentNode: FluxObjectNode
+): FluxObjectNode[] {
+  const dependencyNodes = currentNode.dependsOn
+    .map((dependency) => {
+      const name = dependency.name;
+      let namespace = dependency.namespace;
+
+      if (!namespace) {
+        namespace = currentNode.namespace;
+      }
+
+      return findNode(nodes, name, namespace);
+    })
+    .filter((n) => n);
+
+  const dependentNodes = nodes.filter((node) => {
+    let isDependent = false;
+
+    for (const dependency of node.dependsOn) {
+      const name = dependency.name;
+      let namespace = dependency.namespace;
+      if (!namespace) {
+        namespace = node.namespace;
+      }
+
+      if (name === currentNode.name && namespace === currentNode.namespace) {
+        isDependent = true;
+        break;
+      }
+    }
+
+    return isDependent;
+  });
+
+  return dependencyNodes.concat(dependentNodes);
+}
+
+// getGraphNodes returns all nodes in the current node's dependency tree, including the current node.
```

Comment:
Maybe we could move (some of) these functions to e.g. lib/utils/dependencies.ts? The names are a bit generic when they're in the file `utils` (graph, node and neighbour could all mean different things depending on context).

Reply:
OK, will do it.

Reply:
Done.